source/common/json/json_internal.cc

```diff
@@ -729,6 +729,44 @@ ObjectSharedPtr Factory::loadFromString(const std::string& json) {
   return result.value();
 }
 
+FieldSharedPtr loadFromProtobufValueInternal(const ProtobufWkt::Value& protobuf_value);
+FieldSharedPtr loadFromProtobufStructInternal(const ProtobufWkt::Struct& protobuf_struct);
+
+FieldSharedPtr loadFromProtobufValueInternal(const ProtobufWkt::Value& protobuf_value) {
+  if (protobuf_value.has_string_value()) {
+    return Field::createValue(protobuf_value.string_value());
+  } else if (protobuf_value.has_number_value()) {
+    return Field::createValue(protobuf_value.number_value());
+  } else if (protobuf_value.has_bool_value()) {
+    return Field::createValue(protobuf_value.bool_value());
+  } else if (protobuf_value.has_null_value()) {
+    return Field::createNull();
+  } else if (protobuf_value.has_list_value()) {
+    FieldSharedPtr array = Field::createArray();
+    for (const auto& list_value : protobuf_value.list_value().values()) {
+      array->append(loadFromProtobufValueInternal(list_value));
+    }
+    return array;
+  } else if (protobuf_value.has_struct_value()) {
+    return loadFromProtobufStructInternal(protobuf_value.struct_value());
+  }
+
+  PANIC("not implemented");
```

Comment:
nit: `switch` may be better?

Reply:
Due to 'array' variable declaration in line 745 the switch would require declaring it outside the switch, so I believe if-elseif is cleaner here

Reply:
Sorry, but I didn't got it. You can declaration variable in the switch. I think this is where switch should be used?

Reply:
done